language/evm/move-to-yul/src/context.rs

```diff
@@ -450,6 +450,15 @@ impl<'a> Context<'a> {
         }
     }
 
+    /// Returns true of the type allocates memory.
+    pub fn type_allocates_memory_not_vector(&self, ty: &Type) -> bool {
```

Comment:
Can we use a better name for this? Like `type_is_proper_struct` or something.

Reply:
Thanks! I will fix it in the next PR